pkg/collector/udp.go

```diff
@@ -108,24 +106,21 @@ func (cp *CollectingProcess) startUDPServer() {
 					return
 				}
 				klog.V(2).Infof("Receiving %d bytes from %s", size, address.String())
-				cp.handleUDPClient(address, &wg)
+				cp.handleUDPClient(address)
 				cp.clients[address.String()].packetChan <- bytes.NewBuffer(buff[0:size])
 			}
 		}()
 	}
 	<-cp.stopChan
-	// stop all the workers before closing collector
-	cp.closeAllClients()
-	wg.Wait()
 }
 
-func (cp *CollectingProcess) handleUDPClient(address net.Addr, wg *sync.WaitGroup) {
+func (cp *CollectingProcess) handleUDPClient(address net.Addr) {
 	if _, exist := cp.clients[address.String()]; !exist {
 		client := cp.createClient()
 		cp.addClient(address.String(), client)
-		wg.Add(1)
-		defer wg.Done()
+		cp.wg.Add(1)
 		go func() {
+			defer cp.wg.Done()
```

Comment:
If we are handling UDP server stop as well, we should have a test in UDP tests. Or let's handle it in a different PR?

Reply:
You are right. Added it in current PR since it is simple test. But checking connection openness by reading does not work for udp, so in this PR we only solves the closing of udp.